pkg/operator/observe_config.go

```diff
@@ -84,22 +113,66 @@ func (c ConfigObserver) sync() error {
 	}
 	if len(etcdURLs) > 0 {
 		unstructured.SetNestedStringSlice(observedConfig, etcdURLs, "storageConfig", "urls")
+	} else {
+		glog.Warningf("no etcd endpoints found")
 	}
+	return observedConfig, nil
+}
 
-	// don't worry about errors
-	currentConfig := map[string]interface{}{}
-	json.NewDecoder(bytes.NewBuffer(operatorConfig.Spec.ObservedConfig.Raw)).Decode(&currentConfig)
-	if reflect.DeepEqual(currentConfig, observedConfig) {
-		return nil
+func (c *ConfigObserver) observeClusterConfig(observedConfig map[string]interface{}) (map[string]interface{}, error) {
+
+	// observe CIDRs from cluster-config-v1 in order to populate list of restrictedCIDRs
+
+	clusterConfig, err := c.kubeClient.CoreV1().ConfigMaps("kube-system").Get("cluster-config-v1", metav1.GetOptions{})
+	if err != nil && !errors.IsNotFound(err) {
+		return observedConfig, fmt.Errorf("error retieving cluster config: %s", err)
 	}
 
-	glog.Infof("writing updated observedConfig: %v", diff.ObjectDiff(operatorConfig.Spec.ObservedConfig.Object, observedConfig))
-	operatorConfig.Spec.ObservedConfig = runtime.RawExtension{Object: &unstructured.Unstructured{Object: observedConfig}}
-	if _, err := c.operatorConfigClient.KubeApiserverOperatorConfigs().Update(operatorConfig); err != nil {
-		return err
+	if clusterConfig == nil {
+		glog.Warningf("cluster-config-v1 not found in the kube-system namespace")
+		return observedConfig, nil
 	}
 
-	return nil
+	installConfig := map[string]interface{}{}
+	if installConfigYaml, ok := clusterConfig.Data["install-config"]; ok {
+		if err := yaml.Unmarshal([]byte(installConfigYaml), &installConfig); err != nil {
+			glog.Warningf("Unable to parse install-config: %s", err)
+			return observedConfig, nil
+		}
+	}
+
+	// extract needed values
+	//  data:
+	//   install-config:
+	//     networking:
+	//       podCIDR: 10.2.0.0/16
+	//       serviceCIDR: 10.3.0.0/16
+	restrictedCIDRs := []string{}
+	if networking, ok := installConfig["networking"].(map[string]interface{}); ok {
+		if cidr := networking["podCIDR"]; cidr != nil {
+			restrictedCIDRs = append(restrictedCIDRs, fmt.Sprintf("%v", cidr))
+		} else {
+			glog.Warningf("No value found for install-config/networking/podCIDR.")
```

Comment:
I suspect we will revisit this as a thing to place on status.  Make an issue for us.